GAGE5.py

```diff
@@ -0,0 +1,151 @@
+from Bio import SeqIO
+from Bio.SeqUtils import GC
+from abc import ABC, abstractmethod
+
+def filter_fastq(input_path, gc_bounds=(0, 100), length_bounds=(0, float('inf')), quality_threshold=0, output_filename=None):

```

Comment:
Переменные могут загрустить, что не знают свои типы здесь (но они точно есть в докстринге и это хорошо):

```suggestion
def filter_fastq(input_path: str, gc_bounds: tuple = (0, 100), length_bounds: tuple = (0, float('inf')), quality_threshold: float = 0, output_filename: str = None)->str:
```